File: backend/app/schemas/document.py
from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: str
    doc_number: str
    title: str
    filename: str
    version: str | None = None
    series: str | None = None
    doc_type: str | None = None
    status: str
    word_count: int | None = None
    last_updated: str | None = None
    snippet: str | None = None
    score: float | None = None
    chunk_count: int | None = None
    embedded_chunks: int | None = None

    model_config = {"from_attributes": True}


class DocumentDetail(DocumentResponse):
    content: str
    file_hash: str | None = None


class DocumentList(BaseModel):
    documents: list[DocumentResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/document.ts
export interface Document {
  id: string;
  doc_number: string;
  title: string;
  filename: string;
  version?: string;
  series?: string;
  doc_type?: string;
  status: string;
  word_count?: number;
  last_updated?: string;
  snippet?: string;
  score?: number;
  chunk_count?: number;
  embedded_chunks?: number;
}

export interface DocumentDetail extends Document {
  content: string;
  file_hash?: string;
}

export interface DocumentListResponse {
  documents: Document[];
  total: number;
  page: number;
  page_size: number;
}

export interface DocReference {
  outgoing: { target: string; type: string }[];
  incoming: { doc_number: string; title: string }[];
}
